pkg/container-collection/options.go

```diff
@@ -698,6 +698,9 @@ func WithTracerCollection(tc TracerCollection) ContainerCollectionOption {
 		cc.cacheDelay = 2 * time.Second
 		cc.cachedContainers = &sync.Map{}
 
+		// container.cleanUpFuncs is called either at cc.RemoveContainer() or at cc.Close()
+		var cleanupMu sync.Mutex
```

Comment:
The `cleanUpFuncs` for the same container can be called by two threads in parallel:
1. ContainerCollection's Close() calling WithTracerCollection's cleanUpFuncs calling the Container's cleanUpFuncs.
2. RemoveContainer's AfterFunc() having scheduled 2 seconds ago a call to the Container's cleanUpFuncs

The mutex solution is not very good because it requires cleanup functions to be idempotent and managing mutexes.

Another solution would be to keep the return value of AfterFunc() to be able to cancel the callback with Timer.Stop() but this means we have to store the timers somewhere. Should I do this? Or do you have a better code pattern to use here?

Reply:
I see. I put something together in https://github.com/inspektor-gadget/inspektor-gadget/commit/08d1afcc599144297deef0e814fce43f4b1eb9d3. 

The ideas are:
- Don't use `AfterFunc` but create a single goroutine that cleans up old cached containers. By doing so, we have a single place to acquire a mutex. 
- Remove cleanUpFuncs form container: Even if I like it because is generic, I think we don't need it and that generalization makes the code more difficult to write. 

I haven't tested it too much but I think it can work fine. What do you think?

Reply:
I think that's a lot cleaner! I will pick your patch and test it.